src/gui/codeeditors/qgscodeeditorexpression.cpp

```diff
@@ -36,6 +36,87 @@ Qgis::ScriptLanguage QgsCodeEditorExpression::language() const
   return Qgis::ScriptLanguage::QgisExpression;
 }
 
+void QgsCodeEditorExpression::toggleComment()
+{
+  if ( isReadOnly() )
```

Comment:
This looks like 99% copy and paste from QgsCodeEditorPython::toggleComment. How about moving this to a protected method `QgsCodeEditor::toggleLineComments( const QString& commentPrefix )` instead?